FRC1076-2016-Competition/test/org/usfirst/frc/team1076/test/controller/DistanceAutonomousTest.java

```diff
@@ -72,36 +79,87 @@ public void testVaribleSpeed() {
 	
 	@Test
 	public void testRepeatedMotion() {
-		encoder.reset();
-		
 		DistanceAutonomous auto = new DistanceAutonomous(1.5, 1, encoder);
 		auto.init();
 		auto.driveTrainSpeed();
 		assertEquals(false, auto.shouldChange());
-		encoder.distance = 1.5;
+		encoder.distance += 1.5;
 		assertEquals(true, auto.shouldChange());
 		
 		auto = new DistanceAutonomous(1.5, 1, encoder);
 		auto.init();
 		auto.driveTrainSpeed();
 		assertEquals(false, auto.shouldChange());
-		encoder.distance = 3;
+		encoder.distance += 1.5;
 		auto.driveTrainSpeed();
 		assertEquals(true, auto.shouldChange());
 	}
 	
 	@Test
 	public void testNoArmMotion() {
-		encoder.reset();
 		AutoState auto = new DistanceAutonomous(100, 10, encoder);
 		assertEquals(0, auto.armSpeed(), EPSILON);
 	}
 	
 	@Test
 	public void testNoIntakeMotion() {
-		encoder.reset();
 		AutoState auto = new DistanceAutonomous(100, 10, encoder);
 		assertEquals(0, auto.intakeSpeed(), EPSILON);
 	}
-
+	
+	// DistanceEncoder tests.
+	
+	@Test
+	public void testHighGear() {
+		gear.shiftHigh(robot);
+		AutoState auto = new DistanceAutonomous(120, 10, distanceEncoder);
+		encoder.rawCount += 120 * 34.0/40.0 * 4096 * 6 * Math.PI;
+		distanceEncoder.updateDistance();
+		
+		auto.driveTrainSpeed();
+		assertTrue(auto.shouldChange());
+	}
+	
+	@Test
+	public void testLowGear() {
+		gear.shiftLow(robot);
+		AutoState auto = new DistanceAutonomous(110, 10, distanceEncoder);
+		encoder.rawCount += 110 * 14.0/60.0 * 4096 * 6 * Math.PI;
+		distanceEncoder.updateDistance();
+		
+		auto.driveTrainSpeed();
```

Comment:
also assert the actual distance that the `distanceEncoder` believes it has moved.